src/WP_Export_WXR_Formatter.php

```diff
@@ -17,16 +17,41 @@ public function __construct( $export ) {
 		$this->wxr_version = WXR_VERSION;
 	}
 
-	public function before_posts() {
+	public function before_posts( $opts = [] ) {
 		$before_posts_xml  = '';
-		$before_posts_xml .= $this->header();
-		$before_posts_xml .= $this->site_metadata();
-		$before_posts_xml .= $this->authors();
-		$before_posts_xml .= $this->categories();
-		$before_posts_xml .= $this->tags();
-		$before_posts_xml .= $this->nav_menu_terms();
-		$before_posts_xml .= $this->custom_taxonomies_terms();
-		$before_posts_xml .= $this->rss2_head_action();
+
```

Comment:
Can you deduplicate the code by using an array of `$sections` and then having the logic only once?
```php
$sections = [
    'header',
    'site_metadata',
    ...
];

foreach ( $sections as $section ) {
    if ( ! $opts || in_array( $section, $opts, true ) ) {
        $before_posts_xml .= $this->$section();
    }
}
```

Reply:
Also, the argument is called `$opts`but it only actually represents an array of section names.
We should either rename the argument to `$sections` or allow for arbitrary options but then put the sections under an array key `'sections'` to allow for other options as well.

I think renaming it to `$sections` makes sense here. The above deduplication snippet will need to be changed then to call that array `$available_sections` or similar.

Reply:
done